packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m

```diff
@@ -84,50 +84,49 @@ - (void)downloadManifestFromURL:(NSURL *)url
                                                      timeoutInterval:EXUpdatesDefaultTimeoutInterval];
   [self _setManifestHTTPHeaderFields:request];
   [self _downloadDataWithRequest:request successBlock:^(NSData *data, NSURLResponse *response) {
+      
+    NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse *)response;
+    NSDictionary* headerDictionary = [httpResponse allHeaderFields];
+    NSString* headerSignature = [headerDictionary valueForKey:@"expo-manifest-signature"];
+      
     NSError *err;
     id parsedJson = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
     if (err) {
       errorBlock(err, response);
       return;
     }
 
-    NSDictionary *manifest = [self _extractManifest:parsedJson error:&err];
+    NSDictionary* updateResponseDictionary = [self _extractUpdateResponseDictionary:parsedJson error:&err];
     if (err) {
       errorBlock(err, response);
       return;
     }
 
-    id innerManifestString = manifest[@"manifestString"];
-    id signature = manifest[@"signature"];
-    BOOL isSigned = innerManifestString != nil && signature != nil;
+    id bodyManifestString = updateResponseDictionary[@"manifestString"];
+    id bodySignature = updateResponseDictionary[@"signature"];
+    BOOL isSignatureInBody = bodyManifestString != nil && bodySignature != nil;
 
+    NSString* signature = isSignatureInBody ? bodySignature : headerSignature;
```

Comment:
```suggestion
    id signature = isSignatureInBody ? bodySignature : headerSignature;
```
we don't know yet if either of these are NSStrings

Reply:
good point